runtime/executor/src/test/java/org/apache/nemo/runtime/executor/task/TaskExecutorTest.java

```diff
@@ -330,6 +261,74 @@ public void testParentTaskDataFetching() throws Exception {
     assertTrue(checkEqualElements(elements, runtimeEdgeToOutputData.get(taskOutEdge.getId())));
   }
 
+  /**
+   * The DAG of the task to test will looks like:
+   * source1 -> vertex1 -> vertex2
+   * source2 -> vertex3 ->
+   *
+   * The vertex2 has two incoming edges (from vertex1 and vertex3)
+   * and we test if TaskExecutor handles data and watermarks correctly in this situation.
+   *
+   * The source1 emits watermark 500, 1800 and source2 emits watermark 1000.
+   * The vertex2 should receive and emit watermarks 500 and 1000.
```

Comment:
If watermarks come in this order
source1: 500
source1: 1800
source2: 1000

Shouldn't there be only a single emission of 1000?

Can you comment on the assumptions made in this test, and how the assumptions are ensured?

Reply:
Good point! I will fix this test